app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,15 @@
+class LikesController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    new_like = current_user.likes.new(
+      user_id: current_user.id,
+      post_id: @post.id
+    )
+    new_like.update_likes_counter
+    if new_like.save
+      redirect_to "users/#{@post.user_id}/posts/#{@post.id}", notice: 'Liked!'
+    else
+      redirect_to "/users/#{@post.user_id}/posts/#{@post.id}", alert: 'An error occured!'
+    end
+  end
+end
```

Comment:
- [ ] I received the error below when I tried to like a post.  I fixed this error by modifying your code in the image below and by fixing the typo error you have in the `likes` model. You have `after_save :update_like_counter` instead of having `after_save :update_likes_counter` and I was able to like a post and create a comment on a post too and their counter also gets updated. Kindly implement these changes.


The error | After I modified your code
:---:|:---:
![](https://i.imgur.com/XVGha9s.png) | ![](https://i.imgur.com/tJubYfZ.png)